src/ixdat/plotters/base_mpl_plotter.py

```diff
@@ -1,19 +1,32 @@
 """Base class for plotters using matplotlib"""
 
+from collections import defaultdict
+
 from matplotlib import pyplot as plt
 from matplotlib import gridspec
 
 
 class MPLPlotter:
     """Base class for plotters based on matplotlib. Has methods for making mpl axes."""
 
+    def __init__(self):
+        # Instantiate data holders for dynamic range selection
+        self._dynamically_added_objects = defaultdict(list)
+        self._axis_for_range_selection = set()
+        self._selected_range = {"left": None, "right": None}
+
     def new_ax(self, xlabel=None, ylabel=None):
         """Return a new matplotlib axis optionally with the given x and y labels"""
         fig, ax = plt.subplots()
         if xlabel:
             ax.set_xlabel(xlabel)
         if ylabel:
             ax.set_ylabel(ylabel)
+
+        # Add the axis to those we perform range selection on and connect mouse events
+        self._axis_for_range_selection.add(ax)
+        fig.canvas.mpl_connect("button_press_event", self.onclick)
+
         return ax
 
     def new_two_panel_axes(self, n_bottom=1, n_top=1, emphasis="top"):
```

Comment:
It is because it the mouse event is "connected" to the canvas of the figure, not the axis, so it is actually line 28 that does the magic for this method also, because it is `new_ax` which actually creates the **figure**.

Keyword added.